internal/service/advancedclustertpf/resource.go

```diff
@@ -505,15 +504,34 @@ func resolveTimeout(ctx context.Context, t *timeouts.Value, operationName string
 	return timeoutDuration
 }
 
-func findClusterDiff(ctx context.Context, state, plan *TFModel, diags *diag.Diagnostics, options *update.PatchOptions) (*admin.ClusterDescription20240805, *admin.LegacyAtlasTenantClusterUpgradeRequest, *admin.AtlasTenantClusterUpgradeRequest20240805) {
+type clusterDiff struct {
```

Comment:
```suggestion
type clusterPatchReq struct {
```
can help with readability

Reply:
clusterDiff is really the response from findClusterDiff so I think the name is ok

Reply:
should the name be reconsidered now that findClusterDiff does more things and clusterDiff returns 2 new boolean attributes?

Reply:
I tried to change it to findClusterChanges but i'm not convinced, and also there is some attributeChanges that make it confusing.    I think diff is ok as all info is derived from the difference between the state and the plan, but let me know if you have a suggestion and can try it